src/modules/Authentication.js

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios';
+import store from '../state/store/configureStore';
+import errorHandler from './ErrorHandler';
+
+const Authentication = {
+  async login(event) {
+    let params = {
+      email: event.target.email.value,
+      password: event.target.password.value,
+    };
+
+    axios
+      .post('auth/sign_in', params)
+      .then((createResponse) => {
+        let name = createResponse.data.data.first_name;
+        store.dispatch({
+          type: 'SET_SUBSCRIBE',
+          payload: `Welcome back, ${name}!`,
+        });
+      })
+      .catch((error) => {
+        errorHandler(error);
+      });
+  },
+
+  async subscribe(event, result, setLoading) {
+    if (result.token) {
+      axios
+        .post('/auth', createParams(event))
+        .then((createResponse) => {
+          let name = createResponse.data.data.first_name;
+          let headers = createResponse.headers;
+          axios
+            .post(
```

Comment:
Interesting to know if it is a good practice to do a call inside a call

Reply:
Perhaps not a good practice, but makes sense for this one - we think :D

Reply:
Yes, I'm not 100% sure on this one. I can see and appreciate what you are doing, but let's take a closer look at this in a mob-session, okay?

Reply:
We will go with the current solution